src/main/assets/js/postcodes.ts

```diff
@@ -0,0 +1,126 @@
+
+  constructor() {
+    this.initialize();
+  }
+
+  private initialize(): void {
+    $(() => {
+      if ($(this.tabId).length > 0) {
+        Utilities.toggleTabEnabled(this.postcodesNavTab, false);
+        this.getPostcodes();
+        this.setUpAddEventHandler();
+        this.setUpSelectAllEventHandler();
+        this.setUpDeleteEventHandler();
+        this.setUpMoveEventHandler();
+      }
+    });
+  }
+
+  private getPostcodes(): void {
+    const slug = $('#slug').val();
+
+    $.ajax({
+      url: `/courts/${slug}/postcodes`,
+      method: 'get',
+      success: (res) => {
+        this.updateContent(res);
+      },
+      error: (jqxhr) =>
+        AjaxErrorHandler.handleError(jqxhr, 'GET postcodes failed.')
+    });
+  }
+
+  private setUpAddEventHandler(): void {
+    $(this.tabId).on('click', `button.${this.addPostcodesBtnClass}`, e => {
+      e.preventDefault();
+      const slug = $('#slug').val();
+      const requestBodyData = PostcodesController.getRequestBodyData();
+      requestBodyData['newPostcodes'] = $('[id="addNewPostcodes"]').val();
+
+      $.ajax({
+        url: `/courts/${slug}/postcodes`,
+        method: 'post',
+        data: requestBodyData
+      }).done(res => {
+        this.updateContent(res);
+        window.scrollTo(0, 0);
+      }).fail(response =>
+        AjaxErrorHandler.handleError(response, 'POST new postcodes failed.'));
+    });
+  }
+
+  private setUpMoveEventHandler(): void {
+    $(this.tabId).on('click', `button.${this.movePostcodesBtnClass}`, e => {
+      e.preventDefault();
+      const slug = $('#slug').val();
+      const requestBodyData = PostcodesController.getRequestBodyData();
+      requestBodyData['selectedCourt'] = $('[id="movePostcodesSelect"]').val();
+
+      $.ajax({
+        url: `/courts/${slug}/postcodes`,
+        method: 'put',
+        data: requestBodyData
+      }).done(res => {
+        this.updateContent(res);
+        window.scrollTo(0, 0);
+      }).fail(response =>
+        AjaxErrorHandler.handleError(response, 'PUT (move) postcodes failed.'));
+    });
+  }
+
+  private setUpDeleteEventHandler(): void {
+    $(this.tabId).on('click', `button.${this.deletePostcodesBtnClass}`, e => {
+      e.preventDefault();
+      const slug = $('#slug').val();
+      $.ajax({
+        url: `/courts/${slug}/postcodes`,
+        method: 'delete',
+        data: PostcodesController.getRequestBodyData()
+      }).done(res => {
+        this.updateContent(res);
+        window.scrollTo(0, 0);
+      }).fail(response =>
+        AjaxErrorHandler.handleError(response, 'DELETE postcodes failed.'));
+    });
+  }
+
+  private setUpSelectAllEventHandler(): void {
+    $(this.tabId).on('change', `input[name=${this.selectAllPostcodes}]`, e => {
+      e.preventDefault();
+      // Switch between selecting checked on all (when ticked) and unchecked on all (when not ticked)
+      $('[name="postcodesCheckboxItems"]').prop('checked',
```

Comment:
If you need to use name, make the selector narrower, as discussed. There are several of these in this file.